PowerShellTools/ServiceManagement/ChannelFactoryMaker.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Net.Security;
+using System.ServiceModel;
+using PowerShellTools.Common;
+
+namespace PowerShellTools.ServiceManagement
+{
+    internal static class ChannelFactoryMaker<ServiceType> where ServiceType : class
+    {
+        private static ChannelFactory<ServiceType> _channelFactory;
+        private static object _syncObject = new object();
+
+        public static string EndPointAddress { get; set; }
+
+        public static ChannelFactory<ServiceType> CreateChannelFactory()

```

Comment:
Looks like this factory method is requiring the endpoint address, why not taking it as parameter? Coz consumer of this factory method does not have to know they need initialize a static field before suing this static method....